### PR TITLE
Automatically get and visualize slide dependencies

### PR DESCRIPTION
**Pull request by HeidiSeibold**
See issue #315 

Not sure if `attic` is the right place to put the script. Please let me know what you think.

### CODE REVIEW

**Comment by fabian-s** [comment]
nice!
but ouch :face_with_head_bandage: 

image

would have expected this to look much simpler.... dear god....

File: attic/slide_dependencies.R [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    data.frame(target = nam, depends = unlist(dp), stringsAsFactors = FALSE)
+}
+
+depends <- ldply(names(includes), get_dependencies)
+dependencymat <- na.omit(depends)
+
+dependencymat_simple <- dependencymat
+
+simplify_dp <- function(x) {
+    nams <- gsub(".*([^-]*-[^-]*)-*$", "\\1", x)
```
Comment (fabian-s) [inline attic/slide_dependencies.R]: this collapses different vertices into one, I think, e.g carts-intro and forests-intro..  
just do `gsub("slides-", "", x)` maybe?

Comment (HeidiSeibold) [reply attic/slide_dependencies.R]: That is what we get when we use the non-simple version. I already removed the "slides" part by that point. See lines 8 and 35.

**Comment by fabian-s** [comment]
i think we could merge this now, but would need to spend some more time on the viz and on generating shorter but unique names for the slidesets. maybe better to  leave it open and do it right.

**Comment by HeidiSeibold** [comment]
Can you take care of the unique names and then I'll improve the viz?

**Comment by fabian-s** [comment]
yeah, sure. sorry about deleting your comment, meant to delete my own because i'd seen it was wrong...

**Comment by fabian-s** [comment]
one idea for the viz:  can we do a layout per chapter and then add connections between chapters?

**Comment by fabian-s** [comment]
GEM-Layout:
image

Sugiyama-layout with topic layers:
image

**Comment by HeidiSeibold** [comment]
Nice, thanks @fabian-s :1st_place_medal:

**Comment by HeidiSeibold** [comment]
@fabian-s can we merge this?

**Comment by fabian-s** [comment]
well, you said

> Can you take care of the unique names and then I'll improve the viz?

I did, and also made some progress for the viz, but I don't think this is very useful as is. 
we'd need a topological sort, I think, with all the crossconnections this is way too, or a graph of dependencies between chapters and one within chapters?

i'd rather wait to merge until bernd's refactoring is done, all the slides are reformatted and split into chunks and we have a useable viz as well.

**Comment by HeidiSeibold** [comment]
Ah I thought your viz looks already pretty awesome, which is why I thought it is done. Sorry for the confusion.

**Pull request merged**